src/index.ts

```diff
@@ -54,15 +54,15 @@ export function splitByCase<
     if (previousSplitter === false) {
       // Case rising edge
       if (previousUpper === false && isUpper === true) {
-        parts.push(buff);
+        parts.push(buff.trim());
```

Comment:
I'm wondering if we could fix algorithm that does not add trailing space in first place?